Calculator.js

```diff
@@ -5,71 +5,60 @@ let op1;
 let done;
 
 function cleared() {
-    let screen1 = document.getElementById("screen")
-    screen1.innerHTML = "";
-    screen1.style.padding = "58px"
-    oper = false;
-    op1 = "";
-    done = false;
-
-};
+  let screen1 = document.getElementById("screen");
+  screen1.innerText = "";
+  oper = false;
+  op1 = "";
+  done = false;
+}
 
 function number(num) {
+  if (done) cleared();
 
-    if(done === true) {
-        cleared()
-    }
-
-    let buttn = document.getElementById("screen")
-
-    if (buttn.innerHTML === "Screen") {
-        buttn.innerHTML = "";
-    }
+  let buttn = document.getElementById("screen");
 
-    buttn.style.padding = "33px";
-    buttn.innerHTML += num;
+  if (buttn.innerHTML === "Screen") {
+    buttn.innerHTML = "";
+  }
 
-};
+  buttn.style.padding = "33px";
+  buttn.innerHTML += num;
+}
 
 function operator1(op) {
+  var1 = document.getElementById("screen").innerHTML;
 
-    var1 = document.getElementById("screen").innerHTML
-
-    cleared();
-    op1 = op
-    oper = true;
-
-};
+  cleared();
+  op1 = op;
+  oper = true;
+}
 
 function decim() {
-
-    if(done === true) {
-        cleared();
-    };
-    if (document.getElementById("screen").innerHTML.includes(".")) {
-        document.getElementById("screen").innerHTML = "ERROR!"
-    };
-
-    if(document.getElementById("screen").innerHTML === "") {
-        document.getElementById("screen").innerHTML = "0.";
-    } else {
-        number('.');
-    }
-};
-
-function enter() {
-
-    if (oper === true) {
-        if (op1 === "^") {
-            var2 = document.getElementById("screen").innerHTML;
-            result = var1 ** var2;
-        } else {
-            var2 = document.getElementById("screen").innerHTML; 
-            result = eval(+var1 + op1 + +var2);
-        };
```

Comment:
Eval is generally a security issue but in this case doesn't matter, can always consider a switch statement:

```js
switch (operator) {
    case "+":
        result = +firstOperand + +secondOperand;
        break;
    case "-":
        result = +firstOperand - +secondOperand;
        break;
    case "*":
        result = +firstOperand * +secondOperand;
        break;
    case "/":
        if (+secondOperand === 0) {
            result = "ERROR!"; // Handle division by zero
        } else {
            result = +firstOperand / +secondOperand;
        }
        break;
    case "^":
        result = Math.pow(+firstOperand, +secondOperand);
        break;
    default:
        result = "ERROR!"; // Handle unknown operators
        break;
}

document.getElementById("screen").innerHTML = result;
calculationDone = true;
```